File: llm_generated_code/Gemini/Python/Level1/Q38.py
def max_power(lamps):
    lamps.sort(key=lambda x: x[1], reverse=True)  

    max_a = lamps[0][0]
    total_power = 0
    for a, b in lamps:
        if a <= max_a:
            total_power += b
        else:
            break
    return total_power

for _ in range(int(input())):
    n = int(input())
    lamps = []
    for _ in range(n):
        a, b = map(int, input().split())
        lamps.append((a, b))
    print(max_power(lamps))

File: llm_generated_code/Gemini/CPP/Level1/Q38.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;

        vector<pair<int, int>> lamps(n);
        for (int i = 0; i < n; i++) {
            cin >> lamps[i].first >> lamps[i].second;
        }

        sort(lamps.begin(), lamps.end(), greater<pair<int, int>>());

        int current_power = 0;
        int max_power = 0;

        for (auto lamp : lamps) {
            current_power += lamp.second;
            if (lamp.first <= current_power) {
                max_power = max(max_power, current_power);
            } else {
                break;
            }
        }

        cout << max_power << endl;
    }

    return 0;
}
